terraform-providers/terraform-provider-azurerm
# `azurerm_traffic_manager_profile` - support for new field `max_return` and support for `traffic_routing_method` to be `MultiValue`

👋 Hi all!

This PR adds a new optional field to the `azurerm_traffic_manager_profile` resource, which is `max_return`, which allows profiles with the routing methods `MultiValue` to be set. Without this field, resources like this that attempt to be created are always rejected by the Azure API because the `MaxReturn` value isn't set, so it's impossible to create them with Terraform.

Not coincidentally, this type of routing method was also omitted from the acceptance tests (probably because they wouldn't have passed if `MultiValue` was included). Thus I've done my best to add the missing tests there too, but did run into some trouble running the acceptance tests, so please let me know if there's anything I need to improve there!

Manual tests run by me:
  - Creating a `MultiValue` Traffic Manager **without** `max_return` set, verified we see our new error message complaining about how we need to set it (nicer than getting it from Azure itself)
  - Creating a `MultiValue` Traffic Manager **with** `max_return` set, but with a bad health check config, verified that we see a different error message about that issue instead
  - Creating a `MultiValue` Traffic Manager **with** `max_return` set and everything else correct, verified it can create the Traffic Manger without issue
  - Updated a `MultiValue` Traffic Manager that already had `max_return` set with a new `max_return`, verified it updated correctly on Azure's end

Hope this change is good with you guys, feedback welcome 🙂
